src/Raven.Server/Documents/Compression/CompressedDocument.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Compression;
+
+public class CompressedDocument : Document
+{
+    private readonly DocumentsOperationContext _docsContext;
+    private readonly long _storageId;
+
+    public CompressedDocument(DocumentsOperationContext context, long storageId)
+    {
+        _docsContext = context;
+        _storageId = storageId;
+    }
+
+    public override void Dispose()
+    {
+        if (SkipDispose())
+            return;
+
+        base.Dispose();
+
+        _docsContext.Transaction?.InnerTransaction.ForgetAbout(_storageId);
```

Comment:
wonder if we should just put that inside the `Document` class itself

Reply:
Now you _always_ do `new CompressedDocument()` without even checking if the blittable data is sitting in decompression buffer (I think in previous implementation you had `IsCompressed()`). So yeah, maybe it doesn't  even make sense to introduce `CompressedDocument`

Reply:
dropped `CompressedDocument`, now this logic is inside `Document` class